lib/agent.js

```diff
@@ -198,6 +198,9 @@ function getProxy (proxyUrl, opts, isHttps) {
       return new HttpsProxyAgent(popts)
     }
   } else if (proxyUrl.protocol.startsWith('socks')) {
+    // socks-proxy-agent uses hostname not host
+    popts.hostname = popts.host
+    delete popts.host
```

Comment:
are we removing `host` because it logs a warning if you pass both?

Reply:
I removed it to err on the side of explicitness.  Renaming versus copying.  Otherwise I would have just added `hostname` to popts.  I don't really like that though, that feels like succeeding by coincidence for either agent type.